src/ucp/amo/amo.inl

```diff
@@ -26,25 +26,26 @@ ucs_status_t ucp_amo_check_send_status(ucp_request_t *req, ucs_status_t status);
     UCS_PP_TOKENPASTE(ucp_amo_progress_, _function)
 
 #define UCP_PROGRESS_AMO_DECL(_type, _function, _params) \
-static ucs_status_t _UCP_PROGRESS_AMO_NAME(_function)(uct_pending_req_t *_self) \
+    static ucs_status_t _UCP_PROGRESS_AMO_NAME(_function)(uct_pending_req_t *_self) \
     { \
-    ucp_request_t *req = ucs_container_of(_self, ucp_request_t, send.uct); \
-    ucp_rkey_h rkey    = req->send.amo.rkey; \
-    ucp_ep_t *ep       = req->send.ep; \
-    _type value = (_type)req->send.amo.value;         \
-    _type *result = (_type *)req->send.amo.result;    \
-    uint64_t remote_addr = req->send.amo.remote_addr; \
-    ucs_status_t status; \
-    uct_rkey_t uct_rkey; \
-    uint32_t lane; \
-    \
-    UCP_EP_RESOLVE_RKEY_AMO(ep, rkey, req->send.lane, uct_rkey, \
-                            return UCS_ERR_UNREACHABLE); \
-    lane = req->send.lane; \
-    status = _function(ep->uct_eps[lane], UCS_PP_TUPLE_BREAK _params, remote_addr, \
-                       uct_rkey, result, &req->send.uct_comp); \
-    return ucp_amo_check_send_status(req, status); \
-}
+        ucp_request_t *req = ucs_container_of(_self, ucp_request_t, send.uct); \
+        ucp_rkey_h rkey    = req->send.amo.rkey; \
+        ucp_ep_t *ep       = req->send.ep; \
+        _type value = (_type)req->send.amo.value;         \
+        _type *result = (_type *)req->send.amo.result;    \
+        uint64_t remote_addr = req->send.amo.remote_addr; \
+        ucs_status_t status; \
+        \
+        status = UCP_RKEY_RESOLVE(rkey, ep, amo); \
+        if (status != UCS_OK) { \
+            return UCS_ERR_UNREACHABLE; \
+        } \
+        \
+        req->send.lane = rkey->c.amo_lane; \
+        status = _function(ep->uct_eps[req->send.lane], UCS_PP_TUPLE_BREAK _params, \
+                           remote_addr, rkey->c.amo_rkey, result, &req->send.uct_comp); \
+        return ucp_amo_check_send_status(req, status); \
```

Comment:
note: I would suggest to avoid styling and functional changes in the commit. This would help accelerate review process.